### PR TITLE
Ensure device fallback to `lightning.qubit` successfully carries through for 0 devices

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** To ensure `lightning.gpu` and `lightning.qubit` remain in lock-step for fallback functionality, we must provide better structure around the device validation checks. This PR updates the check and fallback support to always be within a try-except block.

**Description of the Change:** As above.

**Benefits:** Ensures fallback is always caught when required for 0 devices.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane_lightning_gpu/lightning_gpu.py [hunk 1]
```diff
@@ -100,458 +103,471 @@ def _H_dtype(dtype):
...
-                    raise QuantumFunctionError(
-                        "LightningGPU adjoint differentiation method does not currently support the Hermitian observable"
-                    )
-            else:
-                if any([isinstance(o, Projector) for o in m.obs.non_identity_obs]):
-                    raise QuantumFunctionError(
-                        "Adjoint differentiation method does not support the Projector observable"
+        name = "PennyLane plugin for GPU-backed Lightning device using NVIDIA cuQuantum SDK"
+        short_name = "lightning.gpu"
+        pennylane_requires = ">=0.22"
```
Comment (AmintorDusko) [inline pennylane_lightning_gpu/lightning_gpu.py:119]:
```suggestion
        pennylane_requires = ">=0.26"
```

**Review by AmintorDusko — APPROVED** [review]
Hey @mlxd, nice work!
I have just one small suggestion.
Thank you for that!

**Comment by multiphaseCFD** [comment]
Hi @mlxd , nice work!

**Review by multiphaseCFD — APPROVED** [review]

**Comment by AmintorDusko** [comment]
@mlxd, don't forget to fill in the changelog.

**Pull request merged**